src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -463,85 +467,127 @@ private Type EmitIListCoercion<T>(AssociativeNode arg)
             return t;
         }
 
-        private IEnumerable<MethodBase> FunctionLookup(IList args)
+        private IEnumerable<ProtoCore.CLRFunctionEndPoint> FunctionLookup(IList args)
         {
-            IEnumerable<MethodBase> mbs = null;
+            List<ProtoCore.CLRFunctionEndPoint> mbs = new List<ProtoCore.CLRFunctionEndPoint>();
             var key = KeyGen(className, methodName, args.Count);
-            if (methodCache.TryGetValue(key, out IEnumerable<MethodBase> mBase))
+            if (methodCache.TryGetValue(key, out IEnumerable<ProtoCore.CLRFunctionEndPoint> mBase))
             {
-                mbs = mBase;
+                return mBase;
             }
-            else
+
+            int cid = runtimeCore.ClassTable.GetClassId(className);
+            if (!CoreUtils.IsInternalMethod(methodName))
             {
-                if (!CoreUtils.IsInternalMethod(methodName))
+                Validity.Assert(cid != DSASM.Constants.kInvalidIndex);
+            }
+
+            // Can this be used at compile time ? or does this need to be at runtime ? 
+            var fg = runtimeCore.GetFuncGroup(methodName, className);
```

Comment:
Can you explain more what you mean by this comment?

Reply:
The original method (`CallSite::GetFuncGroup`) also handled FunctionGroup lookup by inspecting the arguments (stackValues), which only makes sense at runtime.
I converted the original method for usage at compile time only.

Basically I hope I am not missing any cases by dismissing the arguments analysis.
https://github.com/DynamoDS/Dynamo/blob/master/src/Engine/ProtoCore/Lang/CallSite.cs#L1070

Reply:
@pinzart90 - for methods that are statically typed shouldn't we be able to lookup the correct function group just using types?

Reply:
Oh I see, it's not what I was thinking - if I'm understanding that function correctly it's trying to find the first class in the inheritance chain that matches one of the non array arguments...

We should probably add some tests that try to call methods from one class on another derived object.

Reply:
Looking at the history of this function, it was to implement dynamic dispatch... I agree that is probably not possible at compile time.

Reply:
var pnt = Point.ByCoordinates(0,0,0);
Geometry.Translate(pnt, 1,0,0);

Translate should be found as a Point method not a Geometry method ?
Not sure I understand the use case very well

Reply:
If point implemented its own `Translate` method - but it does not.

There are some examples @aparajit-pratap wrote up for 2.0 release:
https://github.com/DynamoDS/Dynamo/wiki/Dynamo-2.0-Language-Changes-Explained#solved-issues

Reply:
I think long term, we'll need to find some way of both doing the most at compile time, and then falling back to runtime behavior as well, can this function be used in both places.
At compile time - we use this function as you are doing here, and then at runtime we use the version that checks the runtime args to potentially switch to another method?

Reply:
sounds like a good approach to me.